### PR TITLE
Add initial ./src file structure

### PR DESCRIPTION
**Pull request by matthiasbeyer**
This PR adds the initial structure for `./src`, including files which only contain the license header by now.

### CODE REVIEW

**Comment by neithernut** [comment]
ACK.

**Comment by matthiasbeyer** [comment]
The PR changed a bit, as `src/util/attributes.h` was already added. Please review again.

**Comment by neithernut** [comment]
I just noticed that the include-guards are missing from the header files.

**Comment by matthiasbeyer** [comment]
Yes, I don't want to write _any_ source in this PR. And of course, include guards are source!

**Comment by matthiasbeyer** [comment]
If this should be included in this PR, I'll add it... if you want me to!

**Comment by neithernut** [comment]
k, then. I'll open an issue for that.
ACK.

**Comment by matthiasbeyer** [comment]
Saw my latest comment before commenting?

**Comment by neithernut** [comment]
Ah, nay...
Yes, please add the include guards to the PR. It's much cleaner this way.

**Comment by neithernut** [comment]
Could you also add inclusions of the header files from the corresponding source files? (assume `src` as an include directory).

**Comment by matthiasbeyer** [comment]
Nope, because I don't know about the implementation.

I wouldn't like to add it, even if I knew about how things get implemented. I consider it as better if the person who actually implements it has to deal with "what must I include for this".

**Comment by neithernut** [comment]
Let me rephrase that: given a file `a.h` and a file `a.c`. Can you add an inclusion of `a.h` in `a.c`?
We will need _those_ definitely.

**Comment by matthiasbeyer** [comment]
Of course. I will add this.

**Comment by neithernut** [comment]
I prepared a commit. It's availible in my public repository. You can fetch, review and squash it.
(bash ftw.)

**Comment by matthiasbeyer** [comment]
Nice! You scripted it in bash? 

Could you re-do the commit and add a blank line between the `include` and the last comment line? Would be a nicer format, IMHO.

**Comment by neithernut** [comment]
Done.

**Comment by matthiasbeyer** [comment]
I'm ready for squashing, are you?

**Comment by neithernut** [comment]
Yep.

**Comment by matthiasbeyer** [comment]
Done, I wouldn't squash more. Feel free to merge.
